pkg/statedb/sources/common.go

```diff
@@ -0,0 +1,7 @@
+package sources
+
+import "github.com/cilium/cilium/pkg/hive"
+
+type Source[Obj any] interface {
```

Comment:
hmm. `Origin`? `Creator`?

Reply:
`Reflector` is what client-go used, so that's one option as well.